tests/Nox.Types.Extensions.Tests/Types/CurrencyCode/CurrencyCodeTests.cs

```diff
@@ -0,0 +1,134 @@
+    [InlineData("GBP", "British Pound", "GBP", "826", "£")]
+    [InlineData("USD", "US Dollar", "USD", "840", "$")]
+    public void WhenGettingReferenceCurrency_WithValidCurrencyCode_ThenReturnsCurrency(string currencyId, string currencyName, string currencyIsoCode, string currencyIsoNumber, string currencySymbol)
+    {
+        // Arrange
+        var currencyCode = Enum.Parse<CurrencyCode>(currencyIsoCode);
+
+        // Act
+        var referenceCurrency = currencyCode.GetReferenceCurrency();
+
+        // Assert
+        referenceCurrency.Should().NotBeNull();
+        referenceCurrency.Id.Should().Be(currencyId);
+        referenceCurrency.Name.Should().Be(currencyName);
+        referenceCurrency.IsoCode.Should().Be(currencyIsoCode);
+        referenceCurrency.IsoNumber.Should().Be(currencyIsoNumber);
+        referenceCurrency.Symbol.Should().Be(currencySymbol);
+    }
+
+    [Theory]
+    [InlineData("EUR", "Euro", "EUR", "978", "€")]
+    [InlineData("GBP", "British Pound", "GBP", "826", "£")]
+    [InlineData("USD", "US Dollar", "USD", "840", "$")]
+    public void WhenGettingReferenceCurrency_WithValidCurrencyCode3_ThenReturnsCurrency(string currencyId, string currencyName, string currencyIsoCode, string currencyIsoNumber, string currencySymbol)
+    {
+        // Arrange
+        var currencyCode3 = CurrencyCode3.From(currencyIsoCode);
+
+        // Act
+        var referenceCurrency = currencyCode3.GetReferenceCurrency();
+
+        // Assert
+        referenceCurrency.Should().NotBeNull();
+        referenceCurrency.Id.Should().Be(currencyId);
+        referenceCurrency.Name.Should().Be(currencyName);
+        referenceCurrency.IsoCode.Should().Be(currencyIsoCode);
+        referenceCurrency.IsoNumber.Should().Be(currencyIsoNumber);
+        referenceCurrency.Symbol.Should().Be(currencySymbol);
+    }
+
+    [Theory]
+    [InlineData("EUR", "Euro", "EUR", "978", "€")]
+    [InlineData("GBP", "British Pound", "GBP", "826", "£")]
+    [InlineData("USD", "US Dollar", "USD", "840", "$")]
+    public void WhenGettingReferenceCurrency_WithMoneyWithValidCurrencyCode_ThenReturnsCurrency(string currencyId, string currencyName, string currencyIsoCode, string currencyIsoNumber, string currencySymbol)
+    {
+        // Arrange
+        var money = Money.From(1000, Enum.Parse<CurrencyCode>(currencyIsoCode));
+
+        // Act
+        var referenceCurrency = money.GetReferenceCurrency();
+
+        // Assert
+        referenceCurrency.Should().NotBeNull();
+        referenceCurrency.Id.Should().Be(currencyId);
+        referenceCurrency.Name.Should().Be(currencyName);
+        referenceCurrency.IsoCode.Should().Be(currencyIsoCode);
+        referenceCurrency.IsoNumber.Should().Be(currencyIsoNumber);
+        referenceCurrency.Symbol.Should().Be(currencySymbol);
+    }
+
+    [Fact(Skip = "There are 29 missing Nox.Reference.Currency")]
+    public void WhenGettingReferenceCurrency_FromAllCurrencyCodes_Success()
```

Comment:
Following 29 currency codes cannot be found in Nox.Reference:
SSP
SLE
VED
UYW
VES
MRU
STN
ZWL
UYI
CHE
CHW
XBA
XBB
XBC
XBD
XAU
XDR
XAG
XPT
XTS
XPD
XUA
COU
MXV
BOV
CLF
XSU
USN
XXX

Reply:
@dejobratic you can suppress that warning in the code for now. So we can integrate this one.

And meanwhile y should update Nox.Reference and then un skip this tests.

Reply:
This would be covered with https://regusit.atlassian.net/browse/NOX-752
CC: @rochar @dejobratic